java-frontend/src/main/java/org/sonar/java/se/checks/FlowComputation.java

```diff
@@ -117,14 +122,51 @@ private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final
     final ExplodedGraph.Node parent = currentNode.parent();
     learnedConstraints.stream()
       .filter(addToFlow.and(Objects::nonNull))
-      .forEach(lc -> flow.add(location(parent, learnedConstraintMessage(lc, parent))));
+      .forEach(lc -> {
+        flow.add(location(parent, learnedConstraintMessage(lc, currentNode, parent)));
+        if (parent.programPoint.syntaxTree().is(Tree.Kind.METHOD_INVOCATION)) {
+          List<MethodYield> usedMethodYields = currentNode.usedMethodYields(parent);
+          if (!usedMethodYields.isEmpty()) {
```

Comment:
I tend to think that this check should not be required if flowFromMethodInvocation return an empty list.